app-server/app/src/main/scala/footprint/Filter.scala

```diff
@@ -0,0 +1,102 @@
+package com.azavea.rf.footprint
+
+import java.sql.Timestamp
+
+import com.lonelyplanet.akka.http.extensions.{PageRequest, Order}
+import geotrellis.slick.Projected
+import geotrellis.vector.{Point, Polygon}
+
+import com.azavea.rf.datamodel.latest.schema.tables._
+import com.azavea.rf.datamodel.driver.ExtendedPostgresDriver
+import com.azavea.rf.utils.queryparams._
+import com.azavea.rf.utils.UserErrorException
+
+object FootprintFilters {
+  import ExtendedPostgresDriver.api._
+
+  type FootprintsQuery = Query[Footprints, Footprints#TableElementType, Seq]
+
+  val datePart = SimpleFunction.binary[String, Option[Timestamp], Int]("date_part")
+
+  implicit class FootprintDefault[M, U, C[_]](footprints: FootprintsQuery) {
+    def filterByOrganization(orgParams: OrgQueryParameters): FootprintsQuery = {
+      footprints.filter{ footprint =>
+        orgParams.organization.map(footprint.organizationId === _)
+          .reduceLeftOption(_ || _).getOrElse(true:Rep[Boolean])
```

Comment:
How about just `find`?

```scala
footprints
  .filter { footprint =>
    orgParams.organizations.find(_ == footprint.organizationId).isDefined
  }
```

seems a bit inefficient, though, to be doing a find per footprint. Why not have organizations be a Set, then just do a contains, which will be much faster?

```scala
case class OrgQueryParameters(
  organization: Set[UUID] // Note, idiomatically this would be named "organizations"
)
```

```scala
footprints
  .filter { fp => orgParams.organization.contains(fp.organizationId) }
```